include/realizations/catchment/Formulation_Manager.hpp

```diff
@@ -411,19 +412,45 @@ namespace realization {
 
                 // If the directory could be found and opened, we can go ahead and iterate
                 if (directory != nullptr) {
+                    bool match;
                     while ((entry = readdir(directory))) {
-                        // If the entry is a regular file or symlink AND the name matches the pattern, 
-                        //    we can consider this ready to be interpretted as valid forcing data (even if it isn't)
-                        if ((entry->d_type == DT_REG or entry->d_type == DT_LNK) and std::regex_match(entry->d_name, pattern)) {
-                            return forcing_params(
-                                path + entry->d_name,
-                                provider,
-                                simulation_time_config.start_time,
-                                simulation_time_config.end_time
-                            );
-                        }
-                    }
-                }
+                        match = std::regex_match(entry->d_name, pattern);
+                        if( match ) {
+                            // If the entry is a regular file or symlink AND the name matches the pattern, 
+                            //    we can consider this ready to be interpretted as valid forcing data (even if it isn't)
+                            #ifdef _DIRENT_HAVE_D_TYPE
+                            if ( entry->d_type == DT_REG or entry->d_type == DT_LNK ) {
+                                return forcing_params(
+                                    path + entry->d_name,
+                                    provider,
+                                    simulation_time_config.start_time,
+                                    simulation_time_config.end_time
+                                );
+                            }
+                            else if ( entry->d_type == DT_UNKNOWN )
+                            #endif
+                            {
```

Comment:
This is throwing me off - if the _DIRENT_HAVE_D_TYPE condition doesn't pass, does this yield:

```c++
if( match ) {
    {
        struct stat st;
        if( stat((path+entry->d_name)>c_str(), &st) != 0) {...
    }
}
```

Reply:
Yes, it might be a little easier to see from this SS where the IDE does some nice high/low lighting

<img width="873" alt="image" src="https://user-images.githubusercontent.com/5186647/234378420-226fbaf2-257f-4a1c-b66e-98df01669682.png">